idd/Energy+.idd.in

```diff
@@ -4988,15 +4988,15 @@ WindowMaterial:Glazing,
   A5 , \field Window Glass Spectral and Incident Angle Transmittance Data Set Table Name
        \note Used only when Optical Data Type = SpectralAndAngle
        \type object-list
-       \object-list BiVariateTables
+       \object-list TwoDimCurves
```

Comment:
Is that because the form of the data doesn't fit any of our 2D curves?

Reply:
@mjwitte You are right. It requires BiVariateTables only.

Reply:
It's because it uses and manipulates the table data directly [e.g.here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/WindowManager.cc#L849). It was really just used as a pathway to read in the data tables. It never calls `CurveValue`. This may have bigger implementations regarding the refactor.  And if I'm following correctly, AirflowNetwork also uses the table data arrays directly to make some internal tables [here](https://github.com/NREL/EnergyPlus/blob/9655ee264b3c2db52e7cf57fe0d73e331d2e9461/src/EnergyPlus/AirflowNetworkBalanceManager.cc#L5944) but then it uses `CurveValue` to evaluate them later.

Reply:
Not sure. You'll have to work with @lgu1234 to figure that out. It was added [here](https://github.com/NREL/EnergyPlus/pull/6114).

Reply:
@nealkruis @mjwitte After reading data in tables, data manipulation is performed in both functions: SetSameIndeVariableValues and SetCommonIncidentAngles in CurveManager. The table is acted as a holder to allow me to perform data manipulation and resize the table based on inputs from 3 optical properties.

Reply:
@lgu1234 are you okay with me adding sever errors if they are not the right curve types?

Reply:
@mjwitte @lgu1234 I'm pretty sure this is still enforced [here](https://github.com/NREL/EnergyPlus/blob/fb712c7ebe128ce78e7b0b7636645d64f50e1416/src/EnergyPlus/HeatBalanceManager.cc#L2089), just not by the IDD. I'll create a separate issue to expand the types of inputs and eliminate the string comparisons.